src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+
+    if not fpath.is_file():
+        raise argparse.ArgumentTypeError(f"Not a valid file path: {fpath}")
+
+    if fpath.suffix != ".sparql":
+        raise argparse.ArgumentTypeError(f"{fpath} does not have a '.sparql' extension")
+
+    return fpath
+
+
+def check_positive_number(value: str, err_msg: str) -> int:
+    """Ensure 'value' is a positive number.
+
+    Args:
+        value (str): the value to be validated.
+        err_msg (str): used when value fails validation.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated number.
+    """
+    try:
+        number = int(value)
+        if number >= 1:
+            return number
+    except ValueError:
+        pass
+
+    raise argparse.ArgumentTypeError(err_msg)
+
+
+def check_limit(limit: str) -> int:
+    """Validate the 'limit' argument.
+
+    Args:
+        limit (str): the LIMIT to be validated.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated LIMIT
+    """
+    return check_positive_number(
+        limit, "LIMIT must be an integer of value 1 or greater."
+    )
+
+
+def check_timeout(timeout: str) -> int:
+    """Validate the 'timeout' argument.
+
+    Args:
+        timeout (str): the timeout to be validated.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated timeout.
+    """
+    return check_positive_number(
+        timeout, "timeout must be an integer of value 1 or greater."
+    )
+
+
+def main(argv=None) -> int:
+    """The main function.
+
+    Args:
+        argv : If set to None then argparse will use sys.argv as the arguments.
+
+    Returns:
+        int: the exit status - 0: success, any other value - failure.
+    """
+    cli = argparse.ArgumentParser(
+        description=f"run SPARQL queries from the '{PROJECT_ROOT}' project",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    group = cli.add_mutually_exclusive_group(required=True)
+
+    group.add_argument(
+        "-c",
+        "--changed",
+        action="store_true",
+        help="run only changed/new SPARQL queries",
+    )
+
+    group.add_argument(
+        "-a", "--all", action="store_true", help="run all SPARQL queries"
+    )
+
+    group.add_argument(
+        "-f",
+        "--file",
+        help="path to a file containing a valid SPARQL query",
+        type=check_sparql_file,
+    )
+
+    group.add_argument(
+        "-p",
+        "--ping",
+        action="store_true",
+        default=False,
+        help="check responsiveness of endpoint",
+    )
+
+    cli.add_argument(
+        "--timeout",
+        type=check_timeout,
+        default=10,
+        help="maximum number of seconds to wait for a response from the endpoint when 'pinging'",
+    )
+
+    cli.add_argument(
+        "-e",
+        "--endpoint",
+        type=str,
+        default="https://query.wikidata.org/sparql",
+        help="URL of the SPARQL endpoint",
```

Comment:
Even including that we could check other endpoints is so great! As stated, I don't think we'll need to worry about Lexemes moving to a Wikibase anytime soon, but if and when it happens we just switch this and we're good to go!